apis/v1alpha2/gateway_types.go

```diff
@@ -174,7 +174,7 @@ type Listener struct {
 	// Support: Core
 	//
 	// +optional
-	Hostname *Hostname `json:"hostname,omitempty"`
+	Hostname *WildcardHostname `json:"hostname,omitempty"`
```

Comment:
To me, it seems like it would be cleaner to change the type in the filter to `string` and add the custom  validation there. That's a smaller change and lets us keep the (IMHO nicer) `Hostname` typedef for the primary use case.

Reply:
I agree about the smaller change @jpeach might be easier in the end and less user-breaking. 

@robscott does that work for you? I was originally going off of your suggestion of changing to a `WildcardHostname` as in the current commit.

Reply:
We discussed this at the community meeting yesterday, meant to follow up sooner. I think at least @youngnick and I prefer using custom types like `Hostname` and `WildcardHostname` over a string so they're simpler to reuse in the future.

Reply:
In this case, however, it's one place that has custom behavior, so introducing a new type for that seems like a lot. If there are other hostname fields that can't be wildcards, I'd prefer to add the type at that time.

Continuing to rathole on this, what to people think of the type name `PreciseHostname` for a hostname that cannot contain a wildcard label? That's the terminology we use in description, so maybe it makes sense as a type too?

Reply:
I'm sorry to keep bikeshedding this ... `WildcardHostname` implies the name is a wildcard, but it's not. It might be wildcard or precise, so I don't think it's a good choice to replace `Hostname` (which I think is fine).

Reply:
I agree with @jpeach that `Wilcardhostname` does imply it's "only" a wildcard. The `PreciseHostname` might be a good route to use for the redirect field and leave `Hostname` as-is.

Reply:
👍 that works for me.

Reply:
Me too, thanks @stevesloka :)

Reply:
I like the use of `PreciseHostname` as well, that helps to cement that usage as a way of distinguishing them in the code as well as the docs.